rpcs3/Emu/Cell/RawSPUThread.cpp

```diff
@@ -77,10 +89,32 @@ bool RawSPUThread::read_reg(const u32 addr, u32& value)
 		value = status;
 		return true;
 	}
-
+	
+	case Prxy_QueryType_offs:
+	{
+		value = Prxy_QueryType ;
+		return true;
+	}
 	case Prxy_TagStatus_offs:
 	{
-		value = mfc_proxy.size() ? 0 : +mfc_prxy_mask;
+		if (mfc_prxy_mask == 0 || mfc_proxy.size() == 0)
+		{
+			value = mfc_prxy_mask;
+		}
+		
+		else
+		{			
+			u32 completed = mfc_prxy_mask;
+
+			for (u32 i = 0; completed && i < 8; i++)
+			{
+				if (mfc_proxy[i].size)
```

Comment:
like before, cant use the bracket notation here and you need to save the reference initially as it could change between these two calls